le_server.py

```
@@ -258,87 +250,120 @@ def get_hash(self):
+
+    def credentials_are_valid(self, credentials):
+        if 'username' not in credentials.keys():
+            return False
+        if 'plotly_api_domain' not in credentials.keys():
+            return False
+        return True
 
     def user_is_verified(self, credentials):
-        if 'username' not in credentials:
+        if not self.credentials_are_valid(credentials):
             return False
+        # TODO: Decide verification method for on-prem.
+        # https://github.com/plotly/le-bot/issues/2
+        if credentials.get('plotly_api_domain') != 'https://api.plot.ly':
+            return True
         # Continue on knowing there is a username in credentials.
-        response = self.plotly_api(self.plotly_routes.get('current'), credentials)
-        # Requests with either existing or non-existing both return a status_code of 200
+        response = self.call_plotly_api('/v2/users/current', credentials)
+        # Requests with either existing or non-existing both return a
+        # status_code of 200.
         # but non-existing user returns "falsey" values for all parameters.
         content = json.loads(response.content)
         if (response.status_code != 200):
             return False
-        # Continue on knowing the status code was a 200 but username still may not exist.
+        # Continue on knowing the status code was a 200 but username still
+        # may not exist.
         res_usr = content.get('username')
         if (res_usr is not '' and res_usr == credentials.get('username')):
             return True
         else:
             return False
 
-    def delete_certs_folder(self, subdomain):
+    def delete_certs_folder_if_exists(self, subdomain):
         host = self.build_host(subdomain)
         folder_to_delete = self.path_to_certs + host
-        try:
-            shutil.rmtree(folder_to_delete)
-        except Exception as e:
-            print str(e)
+        shutil.rmtree(folder_to_delete, ignore_errors=True)
```

we might want to just do an explicit check if the folder exists, then remove the tree without ignoring errors call `shutil.rmtree(folder_to_delete, ignore_errors=False)`, i.e.
```
if exists(folder_to_delete):
    shutil.rmtree(folder_to_delete)
```

that way, if there are any critical errors that aren't related to the folder existing or not, they get raised and we become aware of them immediately